dubbo-config/dubbo-config-spring/src/main/java/org/apache/dubbo/config/spring/beans/factory/annotation/ReferenceAnnotationBeanPostProcessor.java

```diff
@@ -211,10 +229,19 @@ private String generateReferenceBeanName(AnnotationAttributes attributes, Class<
 
         if (!attributes.isEmpty()) {
             beanNameBuilder.append('(');
-            for (Map.Entry<String, Object> entry : attributes.entrySet()) {
-                beanNameBuilder.append(entry.getKey())
+            //sort attributes keys
+            List<String> sortedAttrKeys = new ArrayList<>(attributes.keySet());
+            Collections.sort(sortedAttrKeys);
+            for (String key : sortedAttrKeys) {
+                Object value = attributes.get(key);
+                //handle method array, generic array
+                if (value != null && value.getClass().isArray()) {
+                    Object[] array = ObjectUtils.toObjectArray(value);
+                    value = Arrays.toString(array);
```

Comment:
It seems that `@Method`'s `@Argument` array is not handled properly.
```
@DubboReference(methods = @Method(name = "sayHello", timeout = 100, arguments = @Argument(index = 0, callback = true)))
```

Some code snippets:

```
for (String key : sortedAttrKeys) {
      Object value = attributes.get(key);
      value = convertAttribute(value);
      beanNameBuilder.append(key)
              .append('=')
              .append(value)
              .append(',');
}
```

```
    private Object convertAttribute(Object obj) {
        if (obj == null) {
            return null;
        }
        if (obj instanceof Annotation) {
            AnnotationAttributes attributes = AnnotationUtils.getAnnotationAttributes((Annotation) obj, true);
            for (Map.Entry<String, Object> entry : attributes.entrySet()) {
                entry.setValue(convertAttribute(entry.getValue()));
            }
            return attributes;
        } else if (obj.getClass().isArray()) {
            Object[] array = ObjectUtils.toObjectArray(obj);
            Object[] newArray = new Object[array.length];
            for (int i = 0; i < array.length; i++) {
                newArray[i] = convertAttribute(array[i]);
            }
            return Arrays.toString(newArray);
        } else {
            return obj;
        }
    }
```

Reply:
Here will handle arrays well. For Example with annotation below:
```
    @DubboReference(version = "2.0", check = false, parameters = {"a", "1"},
            methods = {
                    @Method(name = "sayHello", timeout = 1000, arguments = {
                            @Argument(callback = true, type = "String"),
                            @Argument(callback = false, type = "int")}
                    ),
                    @Method(name = "test", timeout = 3000)
            }
    )
```
will got the follow string 
```
@Reference(check=false,methods=[@org.apache.dubbo.config.annotation.Method(isReturn=true, cache=, deprecated=false, loadbalance=, actives=0, sent=true, timeout=1000, onreturn=, retries=-1, async=false, oninvoke=, sticky=false, arguments=[@org.apache.dubbo.config.annotation.Argument(type=String, callback=true, index=-1), @org.apache.dubbo.config.annotation.Argument(type=int, callback=false, index=-1)], onthrow=, executes=0, merger=, validation=, name=sayHello), @org.apache.dubbo.config.annotation.Method(isReturn=true, cache=, deprecated=false, loadbalance=, actives=0, sent=true, timeout=3000, onreturn=, retries=-1, async=false, oninvoke=, sticky=false, arguments=[], onthrow=, executes=0, merger=, validation=, name=test)],parameters=[a, 1],version=2.0) org.apache.dubbo.demo.DemoServiceA1
```